src/query/sql/src/planner/binder/update.rs

```diff
@@ -102,11 +103,23 @@ impl Binder {
 
             // TODO(zhyass): update_list support subquery.
             let (scalar, _) = scalar_binder.bind(&update_expr.expr).await?;
-            if matches!(scalar, ScalarExpr::SubqueryExpr(_)) {
-                return Err(ErrorCode::Internal(
-                    "update_list in update statement does not support subquery temporarily",
-                ));
+            let f = |scalar: &ScalarExpr| {
+                matches!(
+                    scalar,
+                    ScalarExpr::WindowFunction(_)
+                        | ScalarExpr::AggregateFunction(_)
+                        | ScalarExpr::SubqueryExpr(_)
+                )
+            };
+            let finder = Finder::new(&f);
+            let finder = scalar.accept(finder)?;
+            if !finder.scalars().is_empty() {
+                return Err(ErrorCode::SemanticError(
+                    "update_list in update statement can't contain subquery|window|aggregate functions".to_string(),
```

Comment:
Why can't update_list contain them, is it temporarily unsupported or semantically unsupportable?

Reply:
> temporarily unsupported

subquery is temporarily unsupported. Aggr and window is semantically unsupported